sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1382,3 +1389,44 @@ case class UpCast(child: Expression, dataType: DataType, walkedTypePath: Seq[Str
   extends UnaryExpression with Unevaluable {
   override lazy val resolved = false
 }
+
+object ToBinary {
```

Comment:
can you move it to an individual file?

Reply:
and we need to write unit tests for `ToBinary`. This is the downside of using a custom implementation.

Reply:
Do you suggest trying the approach @maropu proposed, i.e. sharing the `ByteBuffer`? I'm a bit skeptical about the performance implications as you need to reset the buffer etc.

Happy to move the current implementation to a separate file and write unit tests, though.

Reply:
The `ToBinary` doesn't reuse the byte[] as well, it's unfair to reuse ByteBuffer.

Reply:
I moved the conversion logic into `NumberConverter`, an already existing utility object and added tests to the `NumberConverterSuite`.